src/app/components/CharacterDetailRow/index.jsx

```diff
@@ -0,0 +1,100 @@
+import React, { Component } from "react";
+
+export default class CharacterDetailRow extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            editable: false
+        };
+
+        this.updateRow = this.updateRow.bind(this);
+        this.cancelEditing = this.cancelEditing.bind(this);
+    }
+
+    updateRow(e) {
+        this.setState({editable: true});
+    }
+
+    cancelEditing() {
+        this.setState({editable: false});
+    }
+
+    render() {
+        if(this.state.editable) {
+            return (
+                <tr>
+                    <td>
+                        <div className="control">
+                            <input className="input" defaultValue={this.props.data.name}/>
+                        </div>
+                    </td>
+                    <td>
+                        <div className="control">
+                            <input className="input" defaultValue={this.props.data.notation}/>
+                        </div>
+                    </td>
+                    <td>
+                        <div className="control">
```

Comment:
You could probably loop through the props instead to avoid repetition. They all have pretty much the same structure. Only thing different is the "notes" where you do a null check. (could be applied as a general check function)